### PR TITLE
Updates PIN code create and enter screens

### PR DESCRIPTION
**Pull request by amanji**
# Summary of Changes

New PIN create and enter screens using the `react-native-confirmation-code-field` package.
* Introduces PIN input component that replaces the text input fields in PIN create and enter screens
* Adds validation logic to ensure only digits are used for PIN
* Updated wording and alert messages for PIN screens 

# Related Issues

N/A

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [ ] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [ ] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [ ] Updated documentation as needed for changed code and new or modified features;
- [ ] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

**Comment by jleach** [comment]
@PatStLouis Looks like @amanji added place holder values to the French translation file. Able to translate?

**Comment by jleach** [comment]
@marcodrompre Tagging you in case you have time for the translation.

**Review by MosCD3 — COMMENTED** [review]
Don't have write access so just commenting
Approved. Nice work

File: core/App/components/inputs/PinInput.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react'
+import { Keyboard, StyleSheet, Text, TouchableOpacity, View } from 'react-native'
+import { CodeField, useClearByFocusCell } from 'react-native-confirmation-code-field'
+import Icon from 'react-native-vector-icons/MaterialIcons'
+
+import { useTheme } from '../../contexts/theme'
+
+const CELL_COUNT = 6
```
Comment (jleach) [inline core/App/components/inputs/PinInput.tsx]: I think we're using camel case for consts except for keychain/async storage keys.

File: core/App/components/inputs/PinInput.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    <View>
+      {label && <Text style={[TextTheme.label, { marginBottom: 8 }]}>{label}</Text>}
+      <View style={[style.codeField]}>
+        <CodeField
+          {...props}
+          value={pin}
+          onChangeText={(value: string) => {
+            onPinChanged && onPinChanged(value)
+            setPin(value)
+            if (value.length === 6) {
```
Comment (jleach) [inline core/App/components/inputs/PinInput.tsx]: Magic number? Probably meant to be `CELL_COUNT`?

File: core/App/screens/PinCreate.tsx [hunk 3]
```diff
@@ -44,10 +56,49 @@ const PinCreate: React.FC<PinCreateProps> = ({ setAuthenticated }) => {
...
+        title: t('PinCreate.InvalidPIN'),
+        message: t('PinCreate.PleaseUseOnlyNumbersInYourPIN'),
+      })
+    } else if (!x.length) {
+      setModalState({
+        visible: true,
+        title: t('PinCreate.EnterPIN'),
+        message: t('PinCreate.YouNeedToCreateA6DigitPIN'),
+      })
+    } else if (x.length < 6) {
```
Comment (jleach) [inline core/App/screens/PinCreate.tsx]: Magic number.

File: core/App/screens/PinCreate.tsx [hunk 4]
```diff
@@ -44,10 +56,49 @@ const PinCreate: React.FC<PinCreateProps> = ({ setAuthenticated }) => {
...
+        title: t('PinCreate.InvalidPIN'),
+        message: t('PinCreate.PleaseUseOnlyNumbersInYourPIN'),
+      })
+    } else if (!y.length) {
+      setModalState({
+        visible: true,
+        title: t('PinCreate.ReenterPIN'),
+        message: t('PinCreate.PleaseReenterYourPIN'),
+      })
+    } else if (y.length < 6) {
```
Comment (jleach) [inline core/App/screens/PinCreate.tsx]: magic number.

**Review by jleach — CHANGES_REQUESTED** [review]
Looks good. I think the magic number `6` should be moved to a named value like `minimumPinLength` and/or into constants since its used in more than one file.

File: core/App/components/inputs/PinInput.tsx (hunk 1 above)
Comment (amanji) [reply core/App/components/inputs/PinInput.tsx]: Moved to constants file

File: core/App/components/inputs/PinInput.tsx (hunk 2 above)
Comment (amanji) [reply core/App/components/inputs/PinInput.tsx]: Using exported constant

**Review by jleach — DISMISSED** [review]

**Comment by PatStLouis** [comment]
@jleach @amanji We will have a look shortly for the translation

File: core/App/localization/en/index.ts [hunk 5]
```diff
@@ -49,15 +48,24 @@ const translation = {
...
   },
   "PinCreate": {
-    "UserAuthenticationPin": "User authentication pin",
-    "PinMustBe6DigitsInLength": "Pin must be 6 digits in length",
-    "PinsEnteredDoNotMatch": "Pins entered do not match",
-    "6DigitPin": "6 Digit Pin",
-    "ReenterPin": "Re-Enter Pin",
-    "Create": "Create"
+    "UserAuthenticationPIN": "User authentication PIN",
+    "PINTooShort": "PIN too short",
```
Comment (PatStLouis) [inline core/App/localization/en/index.ts:52]:
```suggestion
    "PINTooShort": "PIN too short","PINTooShort": "NIP trop court"
```

File: core/App/localization/en/index.ts [hunk 6]
```diff
@@ -49,15 +48,24 @@ const translation = {
...
+    "InvalidPIN": "Invalid PIN",
```
Comment (PatStLouis) [inline core/App/localization/en/index.ts:53]:
```suggestion
    "InvalidPIN": "Invalid PIN",    "InvalidPIN": "NIP invalide"
```

File: core/App/localization/en/index.ts [hunk 7]
```diff
@@ -49,15 +48,24 @@ const translation = {
...
+    "PINsDoNotMatch": "PINs do not match",
```
Comment (PatStLouis) [inline core/App/localization/en/index.ts:54]:
```suggestion
    "PINsDoNotMatch": "PINs do not match",    "PINsDoNotMatch": "Les NIP ne correspondent pas",
```

File: core/App/localization/en/index.ts [hunk 8]
```diff
@@ -49,15 +48,24 @@ const translation = {
...
+    "PleaseUseOnlyNumbersInYourPIN": "Please use only numbers in your PIN",
```
Comment (PatStLouis) [inline core/App/localization/en/index.ts:55]:
```suggestion
    "PleaseUseOnlyNumbersInYourPIN": "Veuillez utiliser uniquement des chiffres dans votre NIP",
```

File: core/App/localization/en/index.ts [hunk 9]
```diff
@@ -49,15 +48,24 @@ const translation = {
...
+    "YourPINMustBe6DigitsInLength": "Your PIN must be six digits in length",
```
Comment (PatStLouis) [inline core/App/localization/en/index.ts:56]:
```suggestion
    "YourPINMustBe6DigitsInLength": "Votre NIP doit être composé de six chiffres",
```

File: core/App/localization/fr/index.ts [hunk 10]
```diff
@@ -49,15 +47,24 @@ const translation = {
...
+        "PleaseUseOnlyNumbersInYourPIN": "Please use only numbers in your PIN",
+        "YourPINMustBe6DigitsInLength": "Your PIN must be six digits in length",
+        "EnteredPINsDoNotMatch": "The entered PINs do not match. Please try again.",
+        "YouNeedToCreateA6DigitPIN": "You need to create a 6 digit PIN",
+        "PleaseReenterYourPIN": "Please re-enter your PIN",
+        "EnterPIN": "Enter a 6 digit PIN",
+        "ReenterPIN": "Saisir le NIP à nouveau",
+        "CreatePIN": "Create PIN",
+        "RememberPIN": "Remember your PIN.",
+        "PINDisclaimer": "If you forget it, you will need to set up your wallet again and re-add your cards.",
```
Comment (PatStLouis) [inline core/App/localization/fr/index.ts]:
```suggestion
        "UserAuthenticationPIN": "NIP d'authentification de l'utilisateur",
        "PINTooShort": "NIP trop court",
        "InvalidPIN": "NIP invalide",
        "PINsDoNotMatch": "Les NIP ne correspondent pas",
        "PleaseUseOnlyNumbersInYourPIN": "Veuillez utiliser uniquement des chiffres dans votre NIP",
        "YourPINMustBe6DigitsInLength": "Votre NIP doit être composé de six chiffres",
        "EnteredPINsDoNotMatch": "Les NIP saisis ne correspondent pas. Veuillez réessayer.",
        "YouNeedToCreateA6DigitPIN": "Vous devez créer un NIP à 6 chiffres",
        "PleaseReenterYourPIN": "Veuillez saisir à nouveau votre NIP",
        "EnterPIN": "Saisir un NIP à 6 chiffres",
        "ReenterPIN": "Saisir le NIP à nouveau",
        "CreatePIN": "Créer un NIP",
        "RememberPIN": "Mémorisez votre NIP.",
        "PINDisclaimer": "Si vous l'oubliez, vous devrez configurer à nouveau votre portefeuille et ajouter à nouveau vos cartes.",
```

File: core/App/localization/fr/index.ts [hunk 11]
```diff
@@ -49,15 +47,24 @@ const translation = {
...
+      },
     "PinEnter": {
-        "IncorrectPin": "NIP erroné"
+        "IncorrectPIN": "NIP erroné",
+        "EnterPIN": "Please enter your PIN",
```
Comment (PatStLouis) [inline core/App/localization/fr/index.ts]:
```suggestion
        "EnterPIN": "Veuillez saisir votre NIP",
```

**Comment by PatStLouis** [comment]
Here's our suggestions

**Review by jleach — APPROVED** [review]

**Pull request merged**